oshmem/mca/memheap/base/memheap_base_static.c

```diff
@@ -61,7 +65,11 @@ int mca_memheap_base_static_init(mca_memheap_map_t *map)
         size_t total_mem;
 
         for (i = 0, total_mem = 0; i < memheap_context.n_segments; i++) {
-            map_segment_t *s = &map->mem_segs[map->n_segments];
+            map_segment_t *s = mca_memheap_base_allocate_segment(map);
+            if (NULL == s) {
+                MEMHEAP_ERROR("failed to allocate segment");
+                return OSHMEM_ERR_OUT_OF_RESOURCE;
```

Comment:
plz make a cleanup of already allocated ones

Reply:
> plz make a cleanup of already allocated ones

ok, done